### PR TITLE
Port 03 State From TS to JS (ES6 >ES6) Issue #4

### PR DESCRIPTION
**Pull request by linuxonrails**
Port 03 State From TS to JS (ES6 >ES6). This closes Lemoncode/react-by-sample-es6 #4

### CODE REVIEW

File: 03 State/src/app.jsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import * as React from 'react';
+import {HelloComponent} from './hello';
+import {NameEditComponent} from './nameEdit';
+
+interface Props {
+}
+
+interface State {
```
Comment (brauliodiez) [inline 03 State/src/app.jsx]: interface is something valid in ES6?

File: 03 State/src/app.jsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  userName : string;
+}
+
+export class App extends React.Component<Props, State> {
```
Comment (brauliodiez) [inline 03 State/src/app.jsx]: Does this extends <> work fine in Es6?

https://toddmotto.com/react-create-class-versus-component/

File: 03 State/src/app.jsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  constructor(props : Props) {
+    super(props);
+
+    this.state = {userName: 'defaultUserName'};
+  }
+
+  setUsernameState(event) {
+    // If the state gets more complex we should use object.assign
```
Comment (brauliodiez) [inline 03 State/src/app.jsx:17]: Remove this comment, SetState already does that, sorry it's my fault

File: 03 State/src/hello.jsx [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import * as React from 'react';
+
+export const HelloComponent = (props : {userName : string}) => {
```
Comment (brauliodiez) [inline 03 State/src/hello.jsx]: If we are using .js, types should not apply, or are you using flow?

File: 03 State/src/nameEdit.jsx [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import * as React from 'react';
+
+export const NameEditComponent = (props : { userName : string, onChange : (event : any) => any }) => {
```
Comment (brauliodiez) [inline 03 State/src/nameEdit.jsx]: Same thing here, we are using types here, as far as I know ES6 is not typed, and I think we are not using other annotation, does this throws errors when transpiling (babel?)

**Comment by linuxonrails** [comment]
I think I have fixed all problems :-/ ESLint was a great help!

**Comment by brauliodiez** [comment]
Any update on this PR?

**Comment by linuxonrails** [comment]
Everything is alright?

**Comment by brauliodiez** [comment]
I can see.still things like username :string is this valid es6?

**Comment by crsanti** [comment]
I've looked over the changed files and I don't see TS code.

**Comment by brauliodiez** [comment]
Sorry, my fault, looking at the wrong tab, apprving and merging as @vicmator or Rocío to update the issue. Good stuff!!!

**Comment by linuxonrails** [comment]
Thanks!!!

**Pull request merged**